### ISSUE TITLE
Support $XDG_CONFIG_HOME when locating user config files

### ISSUE THREAD

**Issue by HaleTom** [igorshubovych/markdownlint-cli#652]
### Summary

`markdownlint-cli` locates user config files via the `run-con` package (a fork of `rc`). `run-con` computes the home-directory search paths from a **hardcoded** `$HOME/.config` (and `$HOME/.markdownlintrc`, etc.) and never consults the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) `$XDG_CONFIG_HOME` variable. A user who sets `$XDG_CONFIG_HOME` to a custom location gets their `$XDG_CONFIG_HOME/markdownlint/` config silently ignored.

### Steps to reproduce

1. Set `$XDG_CONFIG_HOME` to a non-default location, e.g. `export XDG_CONFIG_HOME=/tmp/xdg-config`.
2. Put a config at `$XDG_CONFIG_HOME/markdownlint/config` (JSON) or `$XDG_CONFIG_HOME/markdownlint/.markdownlintrc` (INI).
3. Run `markdownlint` from a directory with no local `.markdownlintrc`/`.markdownlint.json`.
4. Observe the `$XDG_CONFIG_HOME` config is not loaded.

### Expected behavior

Per the XDG Base Directory Specification, configuration should be read from `$XDG_CONFIG_HOME/markdownlint/` when the variable is set (falling back to `$HOME/.config/markdownlint` when it is not). Currently `run-con` hardcodes `$HOME/.config/`, so an explicit `$XDG_CONFIG_HOME` is ignored.

### Actual behavior

The search paths are computed from `$HOME` directly, so `$XDG_CONFIG_HOME` is ignored. From the [`run-con` source](https://github.com/goatandsheep/rc/blob/main/index.js):

```js
var home = win ? process.env.USERPROFILE : process.env.HOME
// ...
join(home, '.config', name, 'config'),
join(home, '.config', name),
join(home, '.' + name, 'config'),
join(home, '.' + name + 'rc'),
```

The `$HOME/.config/markdownlint` paths are hardcoded and never consult `$XDG_CONFIG_HOME`.

### Environment

- `markdownlint-cli` version: current `master` (0.48.0 / `run-con` 1.3.x)
- Platform: Linux

### Notes / related

- This also affects `markdownlint-cli2`, which has its own config-loading code (`markdownlint-cli2-config`). Filing here first since the user-facing behavior originates in this CLI's `readConfiguration()` path.
- A fix could be applied in `readConfiguration()` (e.g., check `$XDG_CONFIG_HOME/markdownlint/` before the `run-con` `$HOME/.config` fallback) without removing `run-con`, or upstream in `run-con`.

**Comment by DavidAnson** [comment]
As you note, this functionality is provided by the run-con package - please open an issue for that package instead. They are in a much better place to decide whether this suggestion is appropriate or not and to implement it if so.
